usbpd/src/protocol_layer/message/mod.rs

```diff
@@ -269,11 +270,36 @@ impl Message {
             }
         };
 
-        message
+        Ok(message)
     }
 
     /// Parse a message from a slice of bytes.
-    pub fn from_bytes(data: &[u8]) -> Self {
+    pub fn from_bytes(data: &[u8]) -> Result<Self, ParseError> {
         Self::from_bytes_with_state(data, &())
     }
 }
+
+/// Errors that can occur during message/header parsing.
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub enum ParseError {
+    /// The input buffer has an invalid length.
+    /// * `expected` - The expected length.
+    /// * `found` - The actual length found.
+    InvalidLength {
+        /// The expected length.
+        expected: usize,
+        /// The actual length found.
+        found: usize,
+    },
+    /// The specification revision field was invalid.
+    InvalidSpecificationRevision(u8),
```

Comment:
```suggestion
    /// The specification revision is not supported.
    UnsupportedSpecificationRevision(u8),
```

It may happen that this library falls behind the spec at some point.